src/util/psbt/serialize.rs

```diff
@@ -144,3 +155,172 @@ impl Deserialize for SigHashType {
         }
     }
 }
+
+// Taproot related ser/deser
+impl Serialize for schnorr::PublicKey {
+    fn serialize(&self) -> Vec<u8> {
+        self.serialize().to_vec()
```

Comment:
This looks like infinite recursion

Reply:
This works correctly by calling the `serialize()` on `impl schnorr::PublicKey` instead of `psbt::serialize::Serialize`. This is verified by test vectors from the BIP

I will change this to `schnorr::PublicKey::serialize(&self)` to highlight the difference.